### PR TITLE
feat(UIShell): add persistentHamburgerMenu prop to Header

### PR DESCRIPTION
**Pull request by josefaidt**
- adds `persistentHamburgerMenu` prop to Header component
- fixes collapse logic in SideNav
- overrides `isSideNavOpen` logic when persisting hamburger menu

ref #374

### CODE REVIEW

**Review by albertms10 — APPROVED** [review]

File: src/UIShell/SideNav/SideNav.svelte [hunk 1]
```diff
@@ -33,7 +33,7 @@
   class:bx--side-nav="{true}"
   class:bx--side-nav--ux="{true}"
   class:bx--side-nav--expanded="{isOpen}"
-  class:bx--side-nav--collapsed="{!isOpen && fixed}"
+  class:bx--side-nav--collapsed="{fixed ? !isOpen && fixed : !isOpen}"
```
[the remark this answers is not in the document]
Comment (josefaidt) [reply src/UIShell/SideNav/SideNav.svelte:36]: 😆 oh goodness. Applied the simplification

[the remark this answers is not in the document]
Comment (albertms10) [reply src/UIShell/SideNav/SideNav.svelte:36]: Have you already commited this fix? It does not appear as so here!… 🤔

**Review by albertms10 — APPROVED** [review]

**Review by metonym — CHANGES_REQUESTED** [review]
Could you add a [usage example](https://carbon-svelte.vercel.app/components/UIShell#usage)?

File: src/UIShell/SideNav/SideNav.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (josefaidt) [reply src/UIShell/SideNav/SideNav.svelte]: My apologies, just pushed it.

**Review by metonym — APPROVED** [review]

**Pull request merged**